src/mono/mono/mini/interp/transform.c

```diff
@@ -9775,8 +9775,7 @@ mono_interp_transform_method (InterpMethod *imethod, ThreadContext *context, Mon
 
 	MonoJitMemoryManager *jit_mm = get_default_jit_mm ();
 	jit_mm_lock (jit_mm);
-	if (!g_hash_table_lookup (jit_mm->seq_points, imethod->method))
-		g_hash_table_insert (jit_mm->seq_points, imethod->method, imethod->jinfo->seq_points);
+	g_hash_table_replace (jit_mm->seq_points, imethod->method, imethod->jinfo->seq_points);
```

Comment:
I think we also need to do something in `copy_imethod_for_frame`:

In hot reload, the existing version of a method that is already running will finish with the old body, and new calls will use the new method body.  So it is possible to hit a breakpoint in an old version of the method.  So I think we might need to make a copy of the old seq points in `copy_imethod_for_frame`.  What do you think?

Reply:
I'm not sure, this seq_points would only be used to set breakpoint. So it would be possible to be debugging this method with old body?

Like: Thread A is running MethodA and it's stopped in a breakpoint.
In debugger we change something in MethodA so it will replace the MethodA body.
On Thread A we set another breakpoint in another line using the old body and then resume.

Would it be possible to do? While it is stopped in a breakpoint change something and send the deltas to runtime?

Reply:
> Like: Thread A is running MethodA and it's stopped in a breakpoint.
> In debugger we change something in MethodA so it will replace the MethodA body.
> On Thread A we set another breakpoint in another line using the old body and then resume.

I think this is possible.  @tmat - VS can show the old and the new body of MethodA?  Is that something that is already supported by VS, or something we are considering to do in the future?